src/EditorFeatures/CSharpTest/EditAndContinue/ActiveStatementTests.cs

```diff
@@ -2718,6 +2718,81 @@ static void Main(string[] args)
                 Diagnostic(RudeEditKind.UpdateAroundActiveStatement, "lock (G(a => a))", CSharpFeaturesResources.lock_statement));
         }
 
+        [Fact]
+        public void Lock_Update_LocalFunction1()
+        {
+            string src1 = @"
+class C
+{
+    static void Main(string[] args)
+    {
+        void F(int a) {}
+        lock (F)
+        {
+            <AS:0>Console.WriteLine(1);</AS:0>
+        }
+    }
+}
+";
+            string src2 = @"
+class C
+{
+    static void Main(string[] args)
+    {
+        void F(int a) {}
+        lock (F)
+        {
+            <AS:0>Console.WriteLine(2);</AS:0>
+        }
+    }
+}
+";
+            var edits = GetTopEdits(src1, src2);
+            var active = GetActiveStatements(src1, src2);
+
+            edits.VerifyRudeDiagnostics(active,
+                Diagnostic(RudeEditKind.UpdateAroundActiveStatement, "F", CSharpFeaturesResources.local_function)); // To be removed when we will enable EnC for local functions
+        }
+
+        [Fact]
+        public void Lock_Update_LocalFunction2()
+        {
+            string src1 = @"
+class C
+{
+    static void Main(string[] args)
+    {
+        void F(int a) {}
+        void G(int a) {}
+        lock (F)
+        {
+            <AS:0>Console.WriteLine(1);</AS:0>
+        }
+    }
+}
+";
+            string src2 = @"
+class C
+{
+    static void Main(string[] args)
+    {
+        void F(int a) {}
+        void G(int a) {}
+        lock (G)
+        {
+            <AS:0>Console.WriteLine(1);</AS:0>
+        }
+    }
+}
+";
+            var edits = GetTopEdits(src1, src2);
+            var active = GetActiveStatements(src1, src2);
+
+            edits.VerifyRudeDiagnostics(active,
+                Diagnostic(RudeEditKind.UpdateAroundActiveStatement, "F", "local function"), // To be removed when we will enable EnC for local functions
+                Diagnostic(RudeEditKind.UpdateAroundActiveStatement, "lock (G)", CSharpFeaturesResources.lock_statement));
+        }
+
```

Comment:
These are not that useful. The lambda tests were validating that a change within the lock expression that contains lambdas is recognized. Since local functions are separate statements this doesn't apply.